File: Arrays & Strings/#500 - Keyboard Row - Easy/Solution.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
bool isInRow(const char* row, const char* word) {
    for (int i = 0; word[i]; i++) {
        if (!strchr(row, tolower(word[i]))) {
            return false;
        }
    }
    return true;
}

char** findWords(char** words, int wordsSize, int* returnSize) {
    const char* row1 = "qwertyuiop";
    const char* row2 = "asdfghjkl";
    const char* row3 = "zxcvbnm";

    char** result = (char**)malloc(wordsSize * sizeof(char*));
    int count = 0;

    for (int i = 0; i < wordsSize; i++) {
        char* word = words[i];
        if (isInRow(row1, word) || isInRow(row2, word) || isInRow(row3, word)) {
            result[count++] = word;
        }
    }

    *returnSize = count;
    return result;
}

File: Arrays & Strings/#500 - Keyboard Row - Easy/Solution.py
class Solution(object):
    def findWords(self, words):
        """
        :type words: List[str]
        :rtype: List[str]
        """
        row1 = set("qwertyuiop")
        row2 = set("asdfghjkl")
        row3 = set("zxcvbnm")
        
        result = []
        for word in words:
            lower_word = set(word.lower())
            if lower_word.issubset(row1) or lower_word.issubset(row2) or lower_word.issubset(row3):
                result.append(word)
        return result